src/ui/settingsPage/tableElement.js

```diff
@@ -0,0 +1,295 @@
+      font-size: 14px;
+      font-style: normal;
+      font-weight: 400;
+      line-height: 21px;
+    }
+
+    .tableHolder th {
+      background-color: var(--panel-bg-color);
+    }
+    .tableHolder th button {
+      background-color: transparent;
+      border: none;
+      position: relative;
+    }
+    .tableHolder th button.sorted-down::after {
+      content: " ";
+      position: absolute;
+      top: -5px;
+      background-image: url("../../assets/img/arrow-toggle.svg");
+      background-position: center center;
+      background-repeat: no-repeat;
+      margin-inline-start: 8px;
+      pointer-events: none;
+      transform: rotate(0deg);
+      transition: transform 0.275s ease-in-out;
+      inline-size: 24px;
+      height: 24px;
+    }
+    .tableHolder th button.sorted-up::after {
+      content: " ";
+      position: absolute;
+      top: -5px;
+      background-image: url("../../assets/img/arrow-toggle.svg");
+      background-position: center center;
+      background-repeat: no-repeat;
+      margin-inline-start: 8px;
+      pointer-events: none;
+      transform: rotate(180deg);
+      transition: transform 0.275s ease-in-out;
+      inline-size: 24px;
+      height: 24px;
+    }
+    .tableHolder td {
+      padding: 16px;
+    }
+    .deleteBtn {
+      background-color: transparent;
+      color: transparent;
+      border: transparent;
+      background-image: url("../../assets/img/delete-gray.svg");
+      background-position: center center;
+      background-repeat: no-repeat;
+      transition: ease-in-out 0.3s;
+    }
+    .deleteBtn:hover {
+      filter: brightness(1.5);
+    }
+    td.delete {
+      width: 30px;
+    }
+    .row {
+      display: flex;
+      align-content: center;
+      align-items: center;
+    }
+  `;
+}
+customElements.define("mz-context-table", ContextTable);
+
+export const tableHeading = (
+  sortedBy = "",
+  acending = true,
+  sortBy = () => {}
+) => {
+  const getClass = (id) => {
+    if (sortedBy != id) {
+      return "";
+    }
+    if (acending) {
+      return "sorted-down";
+    }
+    return "sorted-up";
+  };
+
+  const onClick = (clickedId) => {
+    // If it's the same, just toggle sort type.
+    if (clickedId === sortedBy) {
+      sortBy(clickedId, !acending);
+      return;
+    }
+    // Otherwise just asc search.
+    sortBy(clickedId, true);
+  };
+
+  return html`
+    <tr class="tableHeader">
+      <th>
+        <button class=${getClass("origin")} @click=${() => onClick("origin")}>
+          Website
+        </button>
+      </th>
+      <th>
+        <button
+          class=${getClass("excluded")}
+          @click=${() => onClick("excluded")}
+        >
+          VPN status
+        </button>
+      </th>
+      <th>
+        <button class=${getClass("city")} @click=${() => onClick("city")}>
+          Location
+        </button>
+      </th>
+      <th></th>
+    </tr>
+  `;
+};
+
+export const noElementPlaceHolder = (contexts = []) => {
```

Comment:
How painful would it be to hide this and just show the empty table when the user is searching and there are no search results?

Reply:
Duh. That is not at all hard. changed it.